### PR TITLE
refactor(jenkins-plugins): convert to class-based datasource

### PR DESCRIPTION
**Pull request by JamieMagee**
## Changes:

- Converts jenkins-plugins datasource to be class-based
- Switches to using Renovate's standard caching instead of custom internal caching

## Context:

#8647

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by JamieMagee** [comment]
I will create and run this on a test repo before merging.

**Review by rarkins — DISMISSED** [review]
LGTM once tested

File: lib/datasource/index.ts
Comment (viceice) [inline lib/datasource/index.ts:217]: Unrelated change?

File: lib/datasource/jenkins-plugins/index.spec.ts
Comment (viceice) [inline lib/datasource/jenkins-plugins/index.spec.ts:21]: I think this whole block can be removed, as here are not jest mocks used.

File: lib/datasource/jenkins-plugins/index.ts
Comment (viceice) [inline lib/datasource/jenkins-plugins/index.ts:6]:
```suggestion
import type {
```

**Review by viceice — COMMENTED** [review]
just some small things 🙃

File: lib/datasource/index.ts
Comment (JamieMagee) [reply lib/datasource/index.ts:217]: Yep 😅 I was investigating replacements in the same branch.

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Comment by JamieMagee** [comment]
Run on https://github.com/renovate-reproductions/12702. Created 4 PRs:

- https://github.com/renovate-reproductions/12702/pull/1
- https://github.com/renovate-reproductions/12702/pull/2
- https://github.com/renovate-reproductions/12702/pull/3
- https://github.com/renovate-reproductions/12702/pull/4

Only made 2 requests to Jenkins API

```json
"hostStats": {
  "api.github.com": {"requestCount": 26, "requestAvgMs": 213, "queueAvgMs": 0},
  "updates.jenkins.io": {
    "requestCount": 2,
    "requestAvgMs": 3688,
    "queueAvgMs": 1
  }
}
```

Followup run made 0 requests to Jenkins API (AKA only used cache)

```json
"hostStats": {"api.github.com": {"requestCount": 17, "requestAvgMs": 451, "queueAvgMs": 0}}
```

**Pull request merged**
